src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequence.tsx

```diff
@@ -132,6 +141,17 @@ const BlastInputSequence = (props: Props) => {
     ? checkSequenceValidity(input, sequenceType)
     : true; // treat empty input as valid
 
+  const { sequences } = useBlastForm();
+  const sequenceValidityKey = index === null ? sequences.length : index;
+  const sequenceValidityContext = useContext(BlastFormContext);
+
+  useEffect(() => {
```

Comment:
Prefer to keep all hooks close to the beginning of the component function. The reason being that you cannot place an if-statement returning null above a hook.

Reply:
Done